FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -143,24 +146,49 @@ - (FIRTrace *)activeTrace {
 }
 
 - (BOOL)isApplicationPreWarmed {
-  NSArray *versionComponents =
-      [[UIDevice currentDevice].systemVersion componentsSeparatedByString:@"."];
-  NSInteger majorVersion = [versionComponents[0] integerValue];
-  if (majorVersion < 15) {
+  if (![self prewarmAvailable]) {
     return NO;
   }
 
-  NSDictionary<NSString *, NSString *> *environment = [NSProcessInfo processInfo].environment;
-  if (environment[@"ActivePrewarm"] != nil &&
-      [environment[@"ActivePrewarm"] isEqualToString:@"1"]) {
-    return YES;
-  }
+  return [self isActivePrewarm] || [self isDoubleDispatchPrewarm];
+}
 
-  if ([doubleDispatchTime compare:applicationFinishLaunchTime] == NSOrderedAscending) {
-    return YES;
+- (void)applyPrewarmTag:(FIRTrace *)trace {
+  BOOL activePrewarm = [self isActivePrewarm];
+  BOOL doubleDispatchPrewarm = [self isDoubleDispatchPrewarm];
+  if (![self prewarmAvailable]) {
+    return;
+  }
+  if (!activePrewarm && !doubleDispatchPrewarm) {
+    [trace setValue:@"cold" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"no" forAttribute:@"is_prewarmed"];
+  } else if (activePrewarm && doubleDispatchPrewarm) {
+    [trace setValue:@"both" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
+  } else if (activePrewarm && !doubleDispatchPrewarm) {
+    [trace setValue:@"active_prewarm" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
+  } else {
+    [trace setValue:@"double_dispatch" forAttribute:@"prewarm_detection"];
+    [trace setValue:@"yes" forAttribute:@"is_prewarmed"];
   }
+}
+
+- (BOOL)prewarmAvailable {
+  NSArray *versionComponents =
```

Comment:
May also check that the platform is iOS. I'm not aware of the prewarm issue is also on tvOS. Also consider using https://github.com/google/GoogleUtilities/blob/main/GoogleUtilities/Environment/third_party/GULAppEnvironmentUtil.m#L245 to get the OS version